server/leafnode.go

```diff
@@ -52,15 +52,15 @@ type leafNodeCfg struct {
 	curURL *url.URL
 }
 
+// Check to see if this is a solicited leafnode. We do special processing for solicited.
 func (c *client) isSolicitedLeafNode() bool {
 	return c.kind == LEAF && c.leaf.remote != nil
 }
 
 // This will spin up go routines to solicit the remote leaf node connections.
 func (s *Server) solicitLeafNodeRemotes(remotes []*RemoteLeafOpts) {
 	for _, r := range remotes {
-		remote := newLeafNodeCfg(r)
-		s.startGoRoutine(func() { s.connectToRemoteLeafNode(remote, true) })
+		s.startGoRoutine(func() { s.connectToRemoteLeafNode(newLeafNodeCfg(r), true) })
```

Comment:
Are you sure we don't need to do it the original way because of `connectToRemoteLeafNode()` being executed as part of a go routine? (you can see `startGoRoutine()` as `go func() {}`) and `r` being a variable of the for loop?

Reply:
I think it is ok but will put it back.